File: backend-spring-boot/src/main/java/com/errormonitoring/dto/ErrorLogDTOs.java
package com.errormonitoring.dto;

import com.errormonitoring.model.ErrorLog;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

public class ErrorLogDTOs {
    
    @Data
    public static class ErrorLogRequest {
        @NotBlank(message = "Application name is required")
        private String applicationName;
        
        @NotBlank(message = "Error message is required")
        private String errorMessage;
        
        private String stackTrace;
        private String source;
        
        @NotBlank(message = "Severity is required")
        private String severity; // Low, Medium, High, Critical
        
        private String status;
        private String environment;
        private String userAgent;
        private String ipAddress;
        private String httpMethod;
        private String apiEndpoint;
        private String errorType;
        private String apiKey; // Optional API key for authentication
    }
    
    @Data
    public static class ErrorLogResponse {
        private Long id;
        private ApplicationDTO application;
        private String endpoint;
        private String errorMessage;
        private String severity;
        private String status;
        private String errorType;
        private String stackTrace;
        private String createdAt;
    }
    
    @Data
    public static class ApplicationDTO {
        private Integer id;
        private String name;
    }
    
    @Data
    public static class ResolveErrorRequest {
        @NotNull(message = "Error ID is required")
        private Integer id;
    }
}

File: backend-error/DTOs/ErrorLogDTOs.cs
namespace ErrorMonitoringAPI.DTOs;

public class ErrorLogDto
{
    public int Id { get; set; }
    public string Application { get; set; } = string.Empty;
    public string? Endpoint { get; set; }
    public string? HttpMethod { get; set; }
    public string ErrorMessage { get; set; } = string.Empty;
    public string? StackTrace { get; set; }
    public string? ErrorType { get; set; }
    public string? ErrorSource { get; set; }
    public string Severity { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public string CreatedAt { get; set; } = string.Empty;
    public string? ResolvedAt { get; set; }
    public int? HttpStatusCode { get; set; }
    public string? UserAgent { get; set; }
    public string? IpAddress { get; set; }
}

public class ErrorLogRequest
{
    public string ApplicationName { get; set; } = string.Empty;
    public string? ApiKey { get; set; }
    public string? ApiEndpoint { get; set; }
    public string? HttpMethod { get; set; }
    public string ErrorMessage { get; set; } = string.Empty;
    public string? StackTrace { get; set; }
    public string? ErrorType { get; set; }
    public string? ErrorSource { get; set; }
    public string Severity { get; set; } = "Medium";
    public int? HttpStatusCode { get; set; }
    public string? UserAgent { get; set; }
    public string? IpAddress { get; set; }
}
